src/writer.rs

```diff
@@ -169,19 +170,41 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
-    /// Write event nested in another level:
-    /// ```xml
-    /// <parent_name>{event}</parent_name>
-    /// ```
-    pub fn write_nested_event<'a, N: AsRef<[u8]>, E: AsRef<Event<'a>>>(
-        &mut self,
-        parent_name: N,
-        event: E,
-    ) -> Result<()> {
-        let name = parent_name.as_ref();
-        self.write_event(Event::Start(BytesStart::borrowed(name, name.len())))?;
-        self.write_event(event)?;
-        self.write_event(Event::End(BytesEnd::borrowed(name)))?;
+    /// Write nested
+    pub fn create_node<'a, N: 'a + AsRef<[u8]>>(&'a mut self, name: &'a N) -> NodeWriter<W> {
+        NodeWriter {
+            writer: self,
+            tag_name: name.as_ref(),
+            start_tag: BytesStart::borrowed_name(name.as_ref()),
```

Comment:
Minor question, unrelated to the patch - what is the use case for ```borrowed``` over ```borrowed_name```?  I also noticed that ```borrowed``` on ```BytesEnd``` behaves like ```borrowed_name``` on ```BytesStart```.

Reply:
`borrowed` will borrow everything (name and attributes) thus you must specify the name length. `borrowed_name` assumes there is no attribute yet. 
For instance, when you're trying to pipe an xml borrowed is great because you can reuse everything. On the other hand when you want to create it from scratch `borrowed_name` is more natural.